client/python/lib/test/test_tensors.py

```diff
@@ -237,96 +353,135 @@ def test_make_tensor_proto_valid_make_ndarray_valid():
-#     values = [[1.0, 2.0, 3.0], [1.0, 2.0, 3.0]]
-#     with pytest.raises(TypeError) as exception_info:
-#         make_tensor_proto(shape = None, dtype = DataType.DT_STRING, values = values)
-#         exception = exception_info.value
-#         assert str(exception) == "[1.0, 2.0, 3.0] has type list, but expected one of: bytes"
+def test_make_tensor_proto_invalid_shape_element_values():
+    values = [1,2,3]
+    with pytest.raises(ValueError) as exception_info:
+        make_tensor_proto(values = values, shape=[-3], dtype = None)
+    exception = exception_info.value
+    assert str(exception) == "shape elements should be unsigned integers"
 
-def test_make_tensor_proto_invalid_dimsions():
+def test_make_tensor_proto_invalid_dimensions():
     values = [[1.0, 2.0], [1.0, 2.0, 3.0]]
     with pytest.raises(ValueError) as exception_info:
-        make_tensor_proto(shape=[2,3], dtype = DataType.DT_FLOAT, values = values)
-        exception = exception_info.value
-        assert str(exception) == "setting an array element with a sequence"
+        make_tensor_proto(values = values, shape=[2,3], dtype = DataType.DT_FLOAT)
+    exception = exception_info.value
+    assert str(exception) == "could not cast values to <class 'numpy.float32'>. setting an array element with a sequence."
 
-def test_make_tensor_proto_invalid_dimsions_2():
-    values = [[(1,2,3)],[(1,2)],[(1,2,3)]]
+def test_make_tensor_proto_invalid_string_to_float_dtype():
+    values = bytes([0x13, 0x00, 0x00, 0x00, 0x08, 0x00])
     with pytest.raises(ValueError) as exception_info:
-        make_tensor_proto(shape=[2,3], dtype = DataType.DT_FLOAT, values = values)
-        exception = exception_info.value
-        assert str(exception) == "setting an array element with a sequence"
+        make_tensor_proto(values = values, shape= None, dtype = DataType.DT_FLOAT)
+    exception = exception_info.value
+    assert str(exception) == "could not cast bytes to <class 'numpy.float32'>. buffer size must be a multiple of element size"
 
-def test_make_tensor_proto_invalid_binary():
-    values = [1,2,3]
-    with pytest.raises(TypeError) as exception_info:
-        make_tensor_proto(shape=None, dtype = DataType.DT_STRING, values = values)
-        exception = exception_info.value
-        assert str(exception) == "1 has type int, but expected one of: bytes"
-
-def test_make_tensor_proto_invalid_shape_type():
-    values = 5.0
-    with pytest.raises(TypeError) as exception_info:
-        make_tensor_proto(shape=0, dtype = DataType.DT_FLOAT, values = values)
-        exception = exception_info.value
-        assert str(exception) == "shape type should be list or tuple, but is int"
+def test_make_tensor_proto_invalid_string_dimensions():
+    values = bytes([0x13, 0x00, 0x00, 0x00, 0x08, 0x00])
+    with pytest.raises(ValueError) as exception_info:
+        make_tensor_proto(values = [[values, values, values],[values, values]], shape= None, dtype = DataType.DT_STRING)
```

Comment:
The issue here is with asarray()? not np.array() call on array creation?
The error message does not make it clear that there's a problem with dimensions.